.gitignore

```diff
@@ -8,3 +8,5 @@ __pycache__/
 .idea
 .vscode/
 version
+.ruff_cache/
+.mypy_cache/
```

Comment:
[nit]: we don't need the .mypy_cache in .gitignore anymore since we're using pyright instead :smile: 

```suggestion
```